glusterd2/transaction/transaction.go

```diff
@@ -13,91 +14,107 @@ import (
 	"github.com/coreos/etcd/clientv3/concurrency"
 	"github.com/pborman/uuid"
 	log "github.com/sirupsen/logrus"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
 )
 
 const (
-	txnPrefix = "transaction/"
+	txnPrefix  = "transaction/"
+	txnTimeOut = time.Second * 15
 )
 
-var expTxn = expvar.NewMap("txn")
+// TxnOptFunc receives a Txn and overrides its members
+type TxnOptFunc func(*Txn) error
 
 // Txn is a set of steps
 type Txn struct {
-	id          uuid.UUID
-	locks       Locks
-	reqID       uuid.UUID
-	storePrefix string
-
-	Ctx             TxnCtx
-	Steps           []*Step
-	DontCheckAlive  bool
-	DisableRollback bool
+	locks Locks
+
 	// Nodes is the union of the all the TxnStep.Nodes and is implicitly
 	// set in Txn.Do(). This list is used to determine liveness of the
 	// nodes before running the transaction steps.
-	Nodes   []uuid.UUID
-	OrigCtx context.Context
+	Nodes           []uuid.UUID `json:"nodes"`
+	StorePrefix     string      `json:"store_prefix"`
+	ID              uuid.UUID   `json:"id"`
+	Locks           []string    `json:"locks"`
+	ReqID           uuid.UUID   `json:"req_id"`
+	Ctx             TxnCtx      `json:"ctx"`
+	Steps           []*Step     `json:"steps"`
+	DontCheckAlive  bool        `json:"dont_check_alive"`
+	DisableRollback bool        `json:"disable_rollback"`
+	Initiator       uuid.UUID   `json:"initiator"`
+	StartTime       time.Time   `json:"start_time"`
+
+	success   chan struct{}
+	error     chan error
```

Comment:
`error` is a package name, can we rename this variable name to something else?

Reply:
`error` here is a member of the `Txn` struct, and doesn't affect the `error` name in the global namespace.